File: web-app-DA/src/app/model/SignUp.ts
export class SignUp {
  name: String;
  username: String;
  email: String;
  password: String;
  roles: String[];

  constructor(name: String, username: String, email: String, password: String) {
    this.name = name;
    this.username = username;
    this.email = email;
    this.password = password;
    this.roles = ['admin'];
  }
}
//
//   get name(): String {
//     return this.name;
//   }
//
//   set name(value: String) {
//     this.name = value;
//   }
//
//   get username(): String {
//     return this.username;
//   }
//
//   set username(value: String) {
//     this.username = value;
//   }
//
//   get email(): String {
//     return this.email;
//   }
//
//   set email(value: String) {
//     this.email = value;
//   }
//
//   get password(): String {
//     return this._password;
//   }
//
//   set password(value: String) {
//     this.password = value;
//   }
//
//   get roles(): String[] {
//     return this.roles;
//   }
//
//   set roles(value: String[]) {
//     this.roles = value;
//   }
// }

File: WebDoAn/src/main/java/com/huy/webdoan/dto/request/SignUp.java
package com.huy.webdoan.dto.request;

import java.util.Set;

public class SignUp {
    private String name;
    private String username;
    private String password;
    private String email;
    private Set<String> roles;

    public SignUp() {
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getUsername() {
        return username;
    }

    public void setUsername(String username) {
        this.username = username;
    }

    public String getPassword() {
        return password;
    }

    public void setPassword(String password) {
        this.password = password;
    }

    public String getEmail() {
        return email;
    }

    public void setEmail(String email) {
        this.email = email;
    }

    public Set<String> getRoles() {
        return roles;
    }

    public void setRoles(Set<String> roles) {
        this.roles = roles;
    }

    public SignUp(String name, String username, String password, String email, Set<String> roles) {
        this.name = name;
        this.username = username;
        this.password = password;
        this.email = email;
        this.roles = roles;
    }
}
